services/base.py

```diff
@@ -440,7 +477,14 @@ def selectImageFromAlbum(self, destinationDir, supportedMimeTypes, displaySize,
       recommendedSize = self.calcRecommendedSize(image['size'], displaySize)
       url = self.addUrlParams(image['url'], recommendedSize, displaySize)
 
-      result = self.requestUrl(url, destination=filename)
+      try:
+        result = self.requestUrl(url, destination=filename)
```

Comment:
Just a side-note, we might want to do this inside the requestUrl function instead and simply return None if the request cannot be handled. This has the benefit that anyone using the requestUrl() call will automatically get the exception handling. But this can be done as a TODO since I'd rather have this massive PR done 😄

Reply:
You're right. I didn't want to do it inside the requestUrl function, because I was not sure if some future service might want to differentiate between different errors (max retries exceeded, request timeout, ...), especially with POST functionallity already built-in! I just didn't want to mess that up, but feel free to change it :)

As for the _"massive PR"_: I'll try to keep my next PRs small and simple ;)